Rg pundit

rayguna opened the pull request:
Set up policies for the `Photos`, `Users`, `Comments`, and `FollowRequests` controllers.

rayguna committed:
Fixed delete button.

rayguna committed:
Update README.md and start working on comments Pundit policy.

rayguna committed:
New comment button is not working.

rayguna committed:
Working on comment policy.

rayguna committed:
Trying to fix the comment class.

rayguna committed:
fixed photo and user policies.

rayguna committed:
Fixed comment and follow_request policies.

rayguna committed:
updated README.md.

rayguna committed:
updated README.md.

rayguna committed:
updated README.md.

rayguna committed:
updated README.md.

rayguna committed:
Update README.md.

rayguna committed:
updated README.md.

kcalva reviewed (approved):
I like how you always update your README to suit your needs and allows you to remember what specifically you did (especially when using pundit gem).

I also think the inheritance from Application_Policy is a very efficient way to clean up your code and it's easier to understand.  Using the before action to authorize actions, well done.

rayguna committed:
Update follow_request_policy.rb

Edited the comment update method to allow either a sender or a recipient to update follow_requests.

ajmompr commented on README.md:
Fabulous job updating your README file; however, this can be made more concise!

ajmompr commented on app/controllers/comments_controller.rb:
```diff
@@ -2,6 +2,8 @@ class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
   before_action :is_an_authorized_user, only: [:destroy, :create]
```
redundant due to before_action { authorize @comment || Comment}

rayguna replied:
Removed. Thanks.

ajmompr commented on app/controllers/comments_controller.rb:
```diff
@@ -70,7 +72,14 @@ def comment_params
     end
 
     def is_an_authorized_user
```
Do we still need this?

rayguna replied:
Removed. Thanks.

ajmompr commented on app/controllers/follow_requests_controller.rb:
```diff
@@ -1,6 +1,8 @@
 class FollowRequestsController < ApplicationController
   before_action :set_follow_request, only: %i[ show edit update destroy ]
 
+  before_action { authorize @follow_request || FollowRequest }
```
good

ajmompr commented on app/controllers/photos_controller.rb:
```diff
@@ -2,7 +2,7 @@ class PhotosController < ApplicationController
   before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
 
-  before_action { authorize @photo }
+  before_action { authorize @photo || Photo}
```
good

ajmompr commented on app/controllers/users_controller.rb:
```diff
@@ -1,7 +1,7 @@
   before_action :set_user, only: %i[ show liked feed followers following discover ]
 
-  before_action { authorize @user }
+  before_action { authorize @user || User}
```
good

ajmompr commented on app/policies/comment_policy.rb:
```diff
@@ -1,3 +1,24 @@
+  end
+
+  def create?
+    true
```
This would allow any user to create a comment on any photo. Who should be able to create a new comment?

ajmompr commented on app/policies/comment_policy.rb:
```diff
@@ -1,3 +1,24 @@
+  end
+
+  def new?
+    true
```
This would allow any user to create a comment on any photo. Who should be able to create a new comment?

rayguna replied:
The constraints are applied by pundit authorization.

ajmompr commented on app/policies/follow_request_policy.rb:
```diff
@@ -1,2 +1,24 @@
+  end
+
+  def new?
+    true
```
refactor to be more DRY

rayguna replied:
Changed. Thanks.

ajmompr commented on app/policies/photo_policy.rb:
what about update and edit?

rayguna replied:
Added. Thanks.

ajmompr commented on app/policies/photo_policy.rb:
```diff
@@ -1,25 +1,28 @@
   end
+
+  def new?
+    true
```
make more DRY

rayguna replied:
Done. Thanks.

ajmompr commented on app/policies/user_policy.rb:
```diff
@@ -1,23 +1,23 @@
+  #methods: show (both for public, follower and followed; and private cases), feed, and discover.
+  
+  def show_photos?
+    record == user.username ||
```
is your feed displaying properly?

ajmompr reviewed (changes requested):
great start; see review

rayguna committed:
edited code as noted in README.md.

rayguna committed:
Fixed the error associated with the edit/delete photo button.

rayguna committed:
Made policy methods more DRY as noted in README.md.

rayguna committed:
fixed the photo edit-delete button issue.

rayguna committed:
updated README.md.
